equipment-service/src/main/java/com/telecominfraproject/wlan/equipment/controller/EquipmentController.java

```diff
@@ -295,6 +297,61 @@ private void checkAllowedChannels(Integer channelNum, String channelType, List<I
 			throw new DsDataValidationException("Equipment contains disallowed " + channelType);
 		}
 	}
+	
+    /**
+     * Updates Equipment Channels
+     *
+     * @param EquipmentChannelsUpdateRequest
+     * @return updated Equipment object
+     * @throws RuntimeException if Equipment record does not exist or if it was modified concurrently
+     */
+    @RequestMapping(value = "/channel", method=RequestMethod.PUT)
+    public Equipment updateChannels(@RequestBody EquipmentChannelsUpdateRequest request) {
+        
+        LOG.debug("updateChannels {}", request);
+        if (request == null) {
+            return null;
+        }
+        Equipment existingEquipment = getOrNull(request.getEquipmentId());
+        
+        if (existingEquipment == null || existingEquipment.getDetails() == null) {
+            LOG.debug("updateChannels: no existing equipment or no details on equipment");
+            return null;
```

Comment:
Should we throw an EntityNotFoundException here to satisfy the return clause in the ServiceInterface ```throws RuntimeException if Equipment record does not exist or if it was modified concurrently```

Reply:
Good point, I will switch to use "get" API instead of "getOrNull" to get existingEquipment